pkg/arvo/ted/naive-csv.hoon

```diff
@@ -0,0 +1,371 @@
+::
+=,  strand=strand:spider
+=,  jael
+::
+^-  thread:spider
+=<  process-logs
+=>
+  |%
+  ::  imported logs is cast as $events
+  +$  events    (list event-log:rpc:ethereum)
+  +$  address   address:naive  :: @ux
+  +$  keccak    @ux            :: used for transaction and roll hashes
+  +$  blocknum  number:block   :: @udblocknumber
+  +$  net       net:dice       ::?(%mainnet %ropsten %local %default)
+  +$  block-map  %+  map  blocknum
+    [timestamp=@da rolls=(map keccak [[gas=@ud sender=address] =effects:naive])]
+  +$  rolls-map  (map blocknum (map keccak effects:naive))
+  ::
+  +$  action  $?  %transfer-point
+                  %spawn
+                  %configure-keys
+                  %escape
+                  %cancel-escape
+                  %adopt
+                  %reject
+                  %detach
+                  %set-management-proxy
+                  %set-spawn-proxy
+                  %set-transfer-proxy
+               ==
+  ::
+  +$  tx-data  $:  block=blocknum
+                   timestamp=@da
+                   roller=address
+                   roll-hash=keccak
+                   tx-hash=keccak
+                   sender=ship
+                   proxy=proxy:naive
+                   nonce=nonce:naive
+                   gas=@ud
+                   length=@ux
+                   suc=?
+                   =action
+                   parent=ship
+                ==
+  --
+::
+|%
+  ::  +process-logs is the main process. it grabs the azimuth snapshop, runs
+  ::  +naive on the logs, grabs the timestamps and gas costs for each roll,
+  ::  then flattens them into a list of $tx-data and saves them to disk.
+  ++  process-logs
+    |=  arg=vase
+    =+  !<([~ =net node-url=@t] arg)
+    =/  pax=path  /naive-exports/csv  :: data will be saved here
+    =/  m  (strand ,vase)
+    ^-  form:m
+    ;<  =events  bind:m  (scry events /gx/azimuth/logs/noun)
+    =/  [naive-contract=address chain-id=@]
+      [naive chain-id]:(get-network:dice net)
+    =/  snap=snap-state:dice  snap
+    ::
+    ;<  ~  bind:m
+      %-  flog-text  %+  weld  "naive-csv: processing {<net>} ethereum logs "
+                               "with {<(lent events)>} events"
+    =/  =rolls-map
+      (compute-effects nas.snap events net naive-contract chain-id)
+    ;<  ~  bind:m  (flog-text "naive-csv: getting timestamps")
+    ;<  tim=thread-result  bind:m
+      %+  await-thread  %eth-get-timestamps
+      !>([node-url ~(tap in ~(key by rolls-map))])
+    =/  timestamps  %-  ~(gas by *(map blocknum @da))
+                    ?-  tim
+                      [%.y *]  ;;((list [@ud @da]) q.p.tim)
+                      [%.n *]  ~|  'naive-csv: %eth-get-timestamps failed'  !!
+                    ==
+    ;<  ~  bind:m  (flog-text "naive-csv: got timestamps")
+    ;<  ~  bind:m  (flog-text "naive-csv: getting tx receipts")
+    ;<  gaz=thread-result  bind:m
+      %+  await-thread  %eth-get-tx-receipts
```

Comment:
Do you mean this for %eth-get-tx-receipts as well since they're both child threads?

Also what's a good way to call a function that lives in a file in `/ted` since there's not a Ford rune to import it (I think)?

Reply:
Probably best to move the function into a lib.  You could also get it with `/=`, which takes any path in the current desk.

Reply:
Yeah I checked and saw `/=` soon after I wrote that, not sure why I thought Ford wouldn't have a rune for that.

You'll have to spell out the logic for moving a thread into a lib for me. I'd think that would make it inaccessible to boilerplate functions like `+await-thread:strandio` without doing something that would be obfuscating. A quick check for `thread` in `/lib` reveals `aqua-vane-thread.hoon`, but its not obvious to me why that shouldn't be in `/ted`.

Reply:
What does +await-thread give you that just calling it as a function with `;<` doesn't?  If it needs to be a thread because another depends on it, then i'd move it into a lib with static arguments, and then the thread just `!<`s the arguments and `!>`s the result